### PR TITLE
done complete finished hooray

### PR DESCRIPTION
**Pull request by keelanpattersoncaesar**

### CODE REVIEW

File: interview-questions.md [hunk 1]
```diff
@@ -7,52 +7,56 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. What is a DOM event?
 
-  Your answer:
+  Your answer: A DOM event is anything the user does while on your React page. It could be hovering a button, changing the page to dark mode, or anything really. Those DOM events change the state of the page. They are constantly updated. 
 
-  Researched answer:
+  Researched answer: A DOM event allows developers to code different triggers when the end user performs an action. This can be used to make webpages extremely fluid and interactive. Basically, when a user performs an action, JS code is executed.
```
Comment (SunkissedQueen) [inline interview-questions.md:20]: Great answer

File: ruby-challenges.rb [hunk 2]
```diff
@@ -31,3 +49,16 @@
 # Expected output: 'LEARN is not a palindrome'
 is_palindrome3 = 'Rotator'
 # Expected output: 'Rotator is a palindrome'
+
+
+def is_or_not(string)
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:54]: Great approach

File: ruby-challenges.rb [hunk 3]
```diff
@@ -10,7 +10,17 @@
 num3 = 221
 # Expected output: '221 is odd'
 
+def even_or_odd(number)
+   if number % 2 === 0
+    p 'even'
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:15]: This is correct logic. Please be mindful of what expected output the question asks for on Lines 7, 9, 11

File: ruby-challenges.rb [hunk 4]
```diff
@@ -21,6 +31,14 @@
 album3 = 'Abbey Road'
 # Expected output: 'bby Rd'
 
+def vowel_snatcher(string)
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:34]: love this method name

File: ruby-challenges.rb [hunk 5]
```diff
@@ -21,6 +31,14 @@
...
+    a = string.delete "aeiou"
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:35]: You definitely got the right approach. Just need to include the uppercase vowels also. Then you will receive the expected output. Great work!

File: interview-questions.md [hunk 6]
```diff
@@ -7,52 +7,56 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: Implicit return means you don't have to specifically tell the program to return a value to you; the return is implied.
 
-  Researched answer:
+  Researched answer: Implicit return means that the last evaluated statement is returned UNLESS you explicitly tell Ruby to return elsewhere. Explicit return can be achieved by using the 'return' keyword. Keep in mind that explicit return can only be used within a method.
```
Comment (SunkissedQueen) [inline interview-questions.md:48]: Awesome research

File: interview-questions.md [hunk 7]
```diff
@@ -7,52 +7,56 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: OOP is a programming that focuses on reusing objects over and over again. It allows developers to easily call on components by name.
 
-  Researched answer:
+  Researched answer: OOP relies on classes and objects. Like I stated before, this programming model reduces ideas into reusable pieces of code which makes the overall workload a lot easier to grasp. Classes are 'blueprints' that can create different objects.
```
Comment (SunkissedQueen) [inline interview-questions.md:30]: What are your thought and research on functional programming?

File: interview-questions.md [hunk 8]
```diff
@@ -7,52 +7,56 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: Props stand for properties. They are used to pass values from parent components to child components.
 
-  Researched answer:
+  Researched answer: Props (properties) are used to pass arguments into React components. Parent => Child. Props are READ ONLY and can only be passed one way.
```
Comment (SunkissedQueen) [inline interview-questions.md:13]: Got to share with the children

File: code-challenges.test.js [hunk 9]
```diff
@@ -20,30 +20,160 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
+// GOOD FAIL:  ReferenceError: dblArrayFn is not defined
+
+// PC:
+// create fn names dblArrayFn
+  // takes in 2 array
+// concat testArray1 to testArray2
+// sort the array
+// for loop over new array 
+  // check for duplicate values with iteration
+    // if value === value at [i], filter it out
```
Comment (SunkissedQueen) [inline code-challenges.test.js:153]: I see you mentioned filter in your pseudocode but I did not see that hof in your actual function below. Consider placing filter in your function as you stated in your pseudocode.

Also the function should not have hardcoded parameters `function dblArrayFn(testArray1, testArray2)`

File: code-challenges.test.js [hunk 10]
```diff
@@ -20,30 +20,160 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
+// use Math.max to find highest value
+  // .push() to empty array 
+// for loop to iterate through array
+// .filter out bad values
+// return placeholder array
+
 const nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
 const nums2 = [109, 5, 9, 67, 8, 24]
 // Expected output: [5, 109]
```
Comment (SunkissedQueen) [inline code-challenges.test.js:106]: https://medium.com/@vladbezden/how-to-get-min-or-max-of-an-array-in-javascript-1c264ec6e1aa

Please review the above link to use the logic of the Math.max and Math.min on your approach. You will see that you can solve this question with less lines.

File: code-challenges.test.js [hunk 11]
```diff
@@ -20,30 +20,160 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
+// // a) Create a test with an expect statement using the variables provided.
 
-// --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
+describe("dblArrayFn", () => {
 
-// a) Create a test with an expect statement using the variables provided.
+  it("that takes in two arrays as arguments and returns one array with no duplicate values", () => {
+    const testArray1 = [3, 7, 10, 5, 4, 3, 3]
+    const testArray2 = [7, 8, 2, 3, 1, 5, 4]
+    expect(dblArrayFn(testArray1, testArray2)).toEqual(expect.arrayContaining([3, 7, 10, 5, 4, 8, 2, 1]))
```
Comment (SunkissedQueen) [inline code-challenges.test.js:140]: that expect.arrayContaining matcher was needed for the first equation. To build efficient tests for the next questions, got back to the normal structure. For instance,

expect(dblArrayFn(testArray1, testArray2)).toEqual(expect.arrayContaining([3, 7, 10, 5, 4, 8, 2, 1]))

Should be as follows:
expect(dblArrayFn(testArray1, testArray2)).toEqual([3, 7, 10, 5, 4, 8, 2, 1])

File: code-challenges.test.js [hunk 12]
```diff
@@ -20,30 +20,160 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
...
+// create fn names dblArrayFn
+  // takes in 2 array
+// concat testArray1 to testArray2
+// sort the array
+// for loop over new array 
+  // check for duplicate values with iteration
+    // if value === value at [i], filter it out
+// return array
+
+// // b) Create the function that makes the test pass
```
Comment (SunkissedQueen) [inline code-challenges.test.js:156]: I am definitely seeing an improvement. Glad that you attempted this question. Please review this link to help you find a solution you understand. https://levelup.gitconnected.com/7-ways-to-remove-duplicates-from-array-in-javascript-cea4144caf31

**Review by SunkissedQueen — APPROVED** [review]
Keelan, you completed the assessment. You need to be proud that you gave this assessment some good effort. I place some suggested links to help guide you to an approach that will give you the output you were trying to obtain. I am proud of the efforts you placed on this assessment.

File: ruby-challenges.rb (hunk 5 above)
Comment (keelanpattersoncaesar) [reply ruby-challenges.rb:35]: Yes, I realized this when we did the assessment! :)

**Pull request merged**